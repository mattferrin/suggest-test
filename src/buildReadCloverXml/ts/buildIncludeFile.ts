import { MeowOutput } from "../../bin/ts/cli";
import { SummaryItem } from "./undefinedOrSummary";

export function buildIncludeFile(input: MeowOutput) {
  return function includeFile(file: SummaryItem | undefined) {
    const isTested =
      (file?.statements ?? 0) > (file?.uncoveredStatements ?? 0) ||
      (file?.conditionals ?? 0) > (file?.uncoveredConditionals ?? 0);
    const isIncludeTested = input.flags.includeTested && isTested;

    const isNotTested =
      (file?.statements ?? 0) === (file?.uncoveredStatements ?? 0) &&
      (file?.conditionals ?? 0) === (file?.uncoveredConditionals ?? 0);
    const isIncludeNotTested = input.flags.includeNotTested && isNotTested;

    return isIncludeTested || isIncludeNotTested;
  };
}
