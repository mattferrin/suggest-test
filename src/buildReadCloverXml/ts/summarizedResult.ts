import { SummaryItem } from "./undefinedOrSummary";
import { Untrusted } from "./Untrusted";

export type FileArray = ReadonlyArray<{
  readonly $: { readonly path: string };
  readonly metrics: ReadonlyArray<{
    readonly $: {
      readonly statements: number;
      readonly coveredstatements: number;
      readonly conditionals: number;
      readonly coveredConditionals: number;
    };
  }>;
  readonly line: ReadonlyArray<{
    readonly $: { readonly type: "cond" };
  }>;
}>;

export function summarizedResult(
  filteredProjectPackageFile: Untrusted<FileArray>
): ReadonlyArray<SummaryItem | undefined> | undefined {
  return filteredProjectPackageFile?.flatMap((projectPackageFile) => {
    const uncoveredStatements =
      (projectPackageFile?.metrics?.[0]?.$?.statements ?? 0) -
      (projectPackageFile?.metrics?.[0]?.$?.coveredstatements ?? 0);
    const statements = projectPackageFile?.metrics?.[0]?.$?.statements ?? 0;

    const uncoveredConditionals =
      (projectPackageFile?.metrics?.[0]?.$?.conditionals ?? 0) -
      (projectPackageFile?.metrics?.[0]?.$?.coveredConditionals ?? 0);
    const conditionals = projectPackageFile?.metrics?.[0]?.$?.conditionals ?? 0;

    return {
      conditionals,
      path: projectPackageFile?.$?.path,
      statements,
      uncoveredConditionals,
      uncoveredStatements,
      weightedConditionals:
        conditionals === 0
          ? 0
          : Math.pow(uncoveredConditionals, 1.3) / conditionals,
      weightedStatements:
        statements === 0 ? 0 : Math.pow(uncoveredStatements, 1.3) / statements,
    };
  });
}
