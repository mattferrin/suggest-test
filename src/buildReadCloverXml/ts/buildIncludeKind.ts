import { MeowOutput } from "../../bin/ts/cli";
import { SummaryItem } from "./undefinedOrSummary";

export function buildIncludeKind(input: MeowOutput) {
  return function includeKind(file: SummaryItem | undefined) {
    const isStatementKeeper =
      (file?.statements ?? 0) > 1 && (file?.uncoveredStatements ?? 0) > 0;
    const isConditionalKeeper =
      (file?.conditionals ?? 0) > 1 && (file?.uncoveredConditionals ?? 0) > 0;

    if (!input.flags.includeConditionals) {
      return isStatementKeeper;
    } else if (!input.flags.includeStatements) {
      return isConditionalKeeper;
    } else {
      return isStatementKeeper || isConditionalKeeper;
    }
  };
}
