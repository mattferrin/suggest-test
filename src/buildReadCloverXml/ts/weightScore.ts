import { MeowOutput } from "../../bin/ts/cli";

export function weightScore(
  input: MeowOutput,
  statements: number,
  uncoveredStatements: number,
  conditionals: number,
  uncoveredConditionals: number
): number {
  const weightedStatements =
    statements === 0 ? 0 : Math.pow(uncoveredStatements, 1.3) / statements;
  const weightedConditionals =
    conditionals === 0
      ? 0
      : Math.pow(uncoveredConditionals, 1.3) / conditionals;

  const weightedBalance = weightedStatements + weightedConditionals;

  if (!input.flags.includeConditionals) {
    return weightedStatements;
  } else if (!input.flags.includeStatements) {
    return weightedConditionals;
  } else {
    return weightedBalance;
  }
}
