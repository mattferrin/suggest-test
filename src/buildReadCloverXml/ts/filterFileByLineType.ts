import { MeowOutput } from "../../bin/ts/cli";
import { FileArray } from "./summarizedResult";
import { Untrusted } from "./Untrusted";

export function filterFileByLineType(
  input: MeowOutput,
  projectPackage: Untrusted<{
    readonly file: FileArray;
  }>
): Untrusted<FileArray> {
  if (!input.flags.includeStatements) {
    return projectPackage?.file?.filter((projectPackageFile) => {
      return projectPackageFile?.line?.some((line) => {
        return line?.$?.type === "cond";
      });
    });
  } else {
    return projectPackage?.file;
  }
}
