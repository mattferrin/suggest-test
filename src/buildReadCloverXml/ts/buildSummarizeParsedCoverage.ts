import { MeowOutput } from "../../bin/ts/cli";
import { filterFileByLineType } from "./filterFileByLineType";
import { FileArray, summarizedResult } from "./summarizedResult";
import { SummaryItem } from "./undefinedOrSummary";
import { Untrusted } from "./Untrusted";

export type Parsed = Untrusted<{
  readonly coverage: {
    readonly project: ReadonlyArray<{
      readonly package: ReadonlyArray<{
        readonly file: FileArray;
      }>;
    }>;
  };
}>;

type SummarizeParsedCoverageOutput =
  | ReadonlyArray<SummaryItem | undefined>
  | undefined;

export function buildSummarizeParsedCoverage(input: MeowOutput) {
  return function summarizeParsedCoverage(
    parsed: Parsed
  ): SummarizeParsedCoverageOutput {
    return parsed?.coverage?.project?.flatMap((project) => {
      return project?.package?.flatMap((projectPackage) => {
        const filteredProjectPackageFile = filterFileByLineType(
          input,
          projectPackage
        );

        return summarizedResult(input, filteredProjectPackageFile);
      });
    });
  };
}
