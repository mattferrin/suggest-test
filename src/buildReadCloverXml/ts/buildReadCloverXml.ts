import fs from "fs/promises";
import { parseStringPromise } from "xml2js";
import { MeowOutput } from "../../bin/ts/cli";
import { buildSortStatements } from "./buildSortStatements";
import { buildSummarizeParsedCoverage } from "./buildSummarizeParsedCoverage";
import { passThroughLogger } from "./passThroughLogger";
import { SummaryItem, undefinedOrSummary } from "./undefinedOrSummary";

export type ReadCloverXmlOutput =
  | { readonly tag: "errors"; readonly errors: readonly Error[] }
  | {
      readonly tag: "summary";
      readonly summaryArray: ReadonlyArray<SummaryItem | undefined>;
      readonly summaryString: string | { readonly tag: "not-ready" };
    };

export function buildReadCloverXml(input: MeowOutput) {
  // eslint-disable-next-line functional/functional-parameters
  return async function readCloverXml(): Promise<ReadCloverXmlOutput> {
    return await fs
      .readFile(input.input[0] ?? "coverage/clover.xml")
      .then(parseStringPromise)
      .then(buildSummarizeParsedCoverage(input))
      .then(undefinedOrSummary)
      .then(buildSortStatements(input))
      .catch((reason): ReadCloverXmlOutput => {
        return { tag: "errors", errors: [new Error(reason)] };
      })
      .then(passThroughLogger);
  };
}
