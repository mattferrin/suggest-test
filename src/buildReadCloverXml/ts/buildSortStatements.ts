import { MeowOutput } from "../../bin/ts/cli";
import { ReadCloverXmlOutput } from "./buildReadCloverXml";
import { summarizeIntoString } from "./summarizeIntoString";

export function buildSortStatements(input: MeowOutput) {
  return function sortStatements(
    output: ReadCloverXmlOutput
  ): ReadCloverXmlOutput {
    switch (output.tag) {
      case "errors":
        return output;
      case "summary": {
        const summaryArray = output.summaryArray
          ?.filter((file) => (file?.statements ?? 0) > 1)
          ?.sort((fileA, fileB) => {
            if (
              fileB?.statements === null ||
              fileB?.statements === undefined ||
              fileA?.statements === null ||
              fileA?.statements === undefined
            ) {
              return -1;
            } else {
              return fileA.weightedStatements > fileB.weightedStatements
                ? -1
                : 1;
            }
          });

        return {
          tag: "summary",
          summaryArray,
          summaryString: summarizeIntoString(summaryArray),
        };
      }
    }
  };
}
