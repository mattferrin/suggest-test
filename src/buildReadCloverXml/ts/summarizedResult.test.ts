import { MeowOutput } from "../../bin/ts/cli";
import { FileArray, summarizedResult } from "./summarizedResult";
import { Untrusted } from "./Untrusted";
import * as weightScore from "./weightScore";

function snap(
  uuid: string,
  comment: string | null,
  input: MeowOutput,
  filteredProjectPackageFile: Untrusted<FileArray>
): void {
  it(uuid, () => {
    /** setup mocks */
    const weightScoreSpy = jest
      .spyOn(weightScore, "weightScore")
      .mockImplementation(() => "weightScore result" as any);

    /** unit under test */
    const result = summarizedResult(input, filteredProjectPackageFile);

    /** assertions */
    expect({
      [`${comment ?? "TODO: comment"}`]: {
        result,
        weightScoreCalls: weightScoreSpy.mock.calls,
      },
    }).toMatchSnapshot();
  });
}

snap(
  "c5353fde-c0eb-45b0-89f0-4a92ae9375a2",
  "`undefined` result from null array",
  "expect as 1st arg to weightScore" as any,
  null
);
snap(
  "7c65727c-f676-4f54-9ab7-60bdf07b5dc1",
  "`undefined` result from undefined array",
  "expect as 1st arg to weightScore" as any,
  undefined
);

snap(
  "b102ad95-daa6-4a7a-b2af-c2f90f31d692",
  "`[]` result from empty array",
  "expect as 1st arg to weightScore" as any,
  []
);

snap(
  "b35725b4-bf32-40fc-ad51-0005b6e67233",
  "null item doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [null]
);
snap(
  "1ab54fed-8e14-434d-9d2d-0d3d96a74c66",
  "undefined item doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [undefined]
);

snap(
  "00b89c24-10ea-4a90-8fed-84f225a31e28",
  "null item metrics doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [{ metrics: null }] as Untrusted<FileArray>
);
snap(
  "bd595fed-1bcf-4190-b611-903dc9caa324",
  "undefined item metrics doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [{ metrics: undefined }] as Untrusted<FileArray>
);

snap(
  "96868bda-df0c-4aa0-a633-1df4c9999b68",
  "null item metrics item doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [{ metrics: [null] }] as Untrusted<FileArray>
);
snap(
  "5a4fbb77-3ea5-4bef-8def-f2c920ab2741",
  "undefined item metrics item doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [{ metrics: [undefined] }] as Untrusted<FileArray>
);

snap(
  "5cbcf1dc-6f60-40b3-8d1c-53c27f36c0e4",
  "null item metrics item $ doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [{ metrics: [{ $: null }] }] as Untrusted<FileArray>
);
snap(
  "a8fc47ff-abaa-4690-89be-0a43eeb94905",
  "undefined item metrics item $ doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [{ metrics: [{ $: undefined }] }] as Untrusted<FileArray>
);

snap(
  "6eac347e-8284-4d08-81a9-0bb4dcf6530c",
  "undefined item metrics item $ propterties don't break things",
  "expect as 1st arg to weightScore" as any,
  [{ metrics: [{ $: {} }] }] as Untrusted<FileArray>
);
snap(
  "403c1ab5-54ff-4a24-8dbf-4fe1e9079161",
  "null item metrics item $ propterties don't break things",
  "expect as 1st arg to weightScore" as any,
  [
    {
      metrics: [
        {
          $: {
            statements: null,
            coveredstatements: null,
            conditionals: null,
            coveredconditionals: null,
          },
        },
      ],
    },
  ] as Untrusted<FileArray>
);

snap(
  "0f20adf1-e300-4055-a718-d44c2915d302",
  "defined item metrics item $ propterties translate properly",
  "expect as 1st arg to weightScore" as any,
  [
    {
      metrics: [
        {
          $: {
            statements: "2",
            coveredstatements: 1,
            conditionals: "6",
            coveredconditionals: 3,
          },
        },
      ],
    },
  ] as Untrusted<FileArray>
);

snap(
  "63be1e68-70b0-4ae6-ba76-8565643a2f49",
  "null item $ path doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [
    {
      $: { path: null },
    },
  ] as Untrusted<FileArray>
);
snap(
  "ff2101ee-7810-4e56-9d8d-cd511eb5e1ff",
  "undefined item $ path doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [
    {
      $: { path: undefined },
    },
  ] as Untrusted<FileArray>
);

snap(
  "4ec6aa85-35a7-4947-a21b-83a1f8fe0d2d",
  "null item $ doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [
    {
      $: null,
    },
  ] as Untrusted<FileArray>
);
snap(
  "7e2816e1-da30-4bd2-a46f-1627baacf368",
  "undefined item $ doesn't break things",
  "expect as 1st arg to weightScore" as any,
  [
    {
      $: undefined,
    },
  ] as Untrusted<FileArray>
);

snap(
  "824dc8e2-5db8-45fa-80ff-7145871c8c9c",
  "defined item $ path translates",
  "expect as 1st arg to weightScore" as any,
  [
    {
      $: { path: "expect as result item path" },
    },
  ] as Untrusted<FileArray>
);
