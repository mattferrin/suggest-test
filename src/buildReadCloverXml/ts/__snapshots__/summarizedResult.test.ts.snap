// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`00b89c24-10ea-4a90-8fed-84f225a31e28 1`] = `
{
  "null item metrics doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`0f20adf1-e300-4055-a718-d44c2915d302 1`] = `
{
  "defined item metrics item $ propterties translate properly": {
    "result": [
      {
        "conditionals": 6,
        "path": undefined,
        "score": "weightScore result",
        "statements": 2,
        "uncoveredConditionals": 3,
        "uncoveredStatements": 1,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        2,
        1,
        6,
        3,
      ],
    ],
  },
}
`;

exports[`1ab54fed-8e14-434d-9d2d-0d3d96a74c66 1`] = `
{
  "undefined item doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`4ec6aa85-35a7-4947-a21b-83a1f8fe0d2d 1`] = `
{
  "null item $ doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`5a4fbb77-3ea5-4bef-8def-f2c920ab2741 1`] = `
{
  "undefined item metrics item doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`5cbcf1dc-6f60-40b3-8d1c-53c27f36c0e4 1`] = `
{
  "null item metrics item $ doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`6eac347e-8284-4d08-81a9-0bb4dcf6530c 1`] = `
{
  "undefined item metrics item $ propterties don't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`7c65727c-f676-4f54-9ab7-60bdf07b5dc1 1`] = `
{
  "\`undefined\` result from undefined array": {
    "result": undefined,
    "weightScoreCalls": [],
  },
}
`;

exports[`7e2816e1-da30-4bd2-a46f-1627baacf368 1`] = `
{
  "undefined item $ doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`63be1e68-70b0-4ae6-ba76-8565643a2f49 1`] = `
{
  "null item $ path doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": null,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`403c1ab5-54ff-4a24-8dbf-4fe1e9079161 1`] = `
{
  "null item metrics item $ propterties don't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`824dc8e2-5db8-45fa-80ff-7145871c8c9c 1`] = `
{
  "defined item $ path translates": {
    "result": [
      {
        "conditionals": 0,
        "path": "expect as result item path",
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`96868bda-df0c-4aa0-a633-1df4c9999b68 1`] = `
{
  "null item metrics item doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`a8fc47ff-abaa-4690-89be-0a43eeb94905 1`] = `
{
  "undefined item metrics item $ doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`b102ad95-daa6-4a7a-b2af-c2f90f31d692 1`] = `
{
  "\`[]\` result from empty array": {
    "result": [],
    "weightScoreCalls": [],
  },
}
`;

exports[`b35725b4-bf32-40fc-ad51-0005b6e67233 1`] = `
{
  "null item doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`bd595fed-1bcf-4190-b611-903dc9caa324 1`] = `
{
  "undefined item metrics doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;

exports[`c5353fde-c0eb-45b0-89f0-4a92ae9375a2 1`] = `
{
  "\`undefined\` result from null array": {
    "result": undefined,
    "weightScoreCalls": [],
  },
}
`;

exports[`ff2101ee-7810-4e56-9d8d-cd511eb5e1ff 1`] = `
{
  "undefined item $ path doesn't break things": {
    "result": [
      {
        "conditionals": 0,
        "path": undefined,
        "score": "weightScore result",
        "statements": 0,
        "uncoveredConditionals": 0,
        "uncoveredStatements": 0,
      },
    ],
    "weightScoreCalls": [
      [
        "expect as 1st arg to weightScore",
        0,
        0,
        0,
        0,
      ],
    ],
  },
}
`;
